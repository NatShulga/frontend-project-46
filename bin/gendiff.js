#!/usr/bin/env node
import { Command } from 'commander';

const program = new Command();
program
  .version('12.0.0')
  .description('The command joins two lines into one')
  .option('-f, --format [type]', 'output format', 'stylish')
  .arguments('<filepath1> <filepath2>')
  .action((filepath1, filepath2, { format }) => {
    console.log(genDiff(filepath1, filepath2, format));
  });
program.parse();

