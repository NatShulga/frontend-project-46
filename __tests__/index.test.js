import path, { dirname } from "path";
import { fileURLToPath } from "url";
import fs from "fs";
import readFile from "../src/utils.js";
import gendiff from "../src/index.js";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const getFixturePath = (filename) => path.resolve(__dirname, "..", "__fixtures__", filename);

const stylish = readFile("stylishFormatTest.txt");
const plain = readFile("plainData.txt");
const json = readFile("jsonData.txt");

describe("testing all formates", () => {
  it("default format", () => {
    expect(
      gendiff(
        getFixturePath("file1.json"),
        getFixturePath("file2.yaml"),
        "stylish"
      )
    ).toEqual(stylish);
  });
  it("stylish format, json files", () => {
      const result = gendiff(
        getFixturePath("file1.json"),
        getFixturePath("file2.json")
      );
      expect(result).toEqual(stylish)
  });
  it("stylish format, yaml files, json files", () => {
      const result = gendiff(getFixturePath("file1.yaml"),
      getFixturePath("file2.yaml"))
      expect(result).toEqual(stylish)
  });
  it("plain format, json files", () => {
    expect(
      gendiff(
        getFixturePath("file1.json"),
        getFixturePath("file2.json"),
        "plain"
      )
    ).toEqual(plain);
  });
  it("plain format, yaml files", () => {
    expect(
      gendiff(
        getFixturePath("file1.yaml"),
        getFixturePath("file2.yaml"),
        "plain"
      )
    ).toEqual(plain);
  });
  
  it("json format, json files, yaml files", () => {
    const json = []
    const result = gendiff(
      getFixturePath("file1.json"),
      getFixturePath("file2.json"),
      "json"
    );
  
    expect(result).toEqual(json);
  });

  it("json format, yaml files", () => {
    const result = gendiff(getFixturePath("file1.yaml"), getFixturePath("file2.yaml"));
    expect(result).toEqual(stylish);
  });
});
