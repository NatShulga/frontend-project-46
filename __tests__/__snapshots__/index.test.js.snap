// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing all formates json format, json files, yaml files 1`] = `
"[
 {
  "type": "nested",
  "key": "common",
  "children": [
   {
    "type": "added",
    "key": "follow",
    "value": false
   },
   {
    "type": "unchanged",
    "key": "setting1",
    "value": "Value 1"
   },
   {
    "type": "removed",
    "key": "setting2",
    "value": 200
   },
   {
    "type": "changed",
    "key": "setting3",
    "value1": true,
    "value2": null
   },
   {
    "type": "added",
    "key": "setting4",
    "value": "blah blah"
   },
   {
    "type": "added",
    "key": "setting5",
    "value": {
     "key5": "value5"
    }
   },
   {
    "type": "nested",
    "key": "setting6",
    "children": [
     {
      "type": "nested",
      "key": "doge",
      "children": [
       {
        "type": "changed",
        "key": "wow",
        "value1": "",
        "value2": "so much"
       }
      ]
     },
     {
      "type": "unchanged",
      "key": "key",
      "value": "value"
     },
     {
      "type": "added",
      "key": "ops",
      "value": "vops"
     }
    ]
   }
  ]
 },
 {
  "type": "nested",
  "key": "group1",
  "children": [
   {
    "type": "changed",
    "key": "baz",
    "value1": "bas",
    "value2": "bars"
   },
   {
    "type": "unchanged",
    "key": "foo",
    "value": "bar"
   },
   {
    "type": "changed",
    "key": "nest",
    "value1": {
     "key": "value"
    },
    "value2": "str"
   }
  ]
 },
 {
  "type": "removed",
  "key": "group2",
  "value": {
   "abc": 12345,
   "deep": {
    "id": 45
   }
  }
 },
 {
  "type": "added",
  "key": "group3",
  "value": {
   "deep": {
    "id": {
     "number": 45
    }
   },
   "fee": 100500
  }
 }
]"
`;

exports[`testing all formates json format, json files, yaml files: [
 {
  "key": "common",
  "children": [
   {
    "key": "follow",
    "value": false,
    "type": "added"
   },
   {
    "key": "setting1",
    "value": "Value 1",
    "type": "unchanged"
   },
   {
    "key": "setting2",
    "value": 200,
    "type": "removed"
   },
   {
    "key": "setting3",
    "oldValue": true,
    "newValue": null,
    "type": "updated"
   },
   {
    "key": "setting4",
    "value": "blah blah",
    "type": "added"
   },
   {
    "key": "setting5",
    "value": {
     "key5": "value5"
    },
    "type": "added"
   },
   {
    "key": "setting6",
    "children": [
     {
      "key": "doge",
      "children": [
       {
        "key": "wow",
        "oldValue": "",
        "newValue": "so much",
        "type": "updated"
       }
      ],
      "type": "nested"
     },
     {
      "key": "key",
      "value": "value",
      "type": "unchanged"
     },
     {
      "key": "ops",
      "value": "vops",
      "type": "added"
     }
    ],
    "type": "nested"
   }
  ],
  "type": "nested"
 },
 {
  "key": "group1",
  "children": [
   {
    "key": "baz",
    "oldValue": "bas",
    "newValue": "bars",
    "type": "updated"
   },
   {
    "key": "foo",
    "value": "bar",
    "type": "unchanged"
   },
   {
    "key": "nest",
    "oldValue": {
     "key": "value"
    },
    "newValue": "str",
    "type": "updated"
   }
  ],
  "type": "nested"
 },
 {
  "key": "group2",
  "value": {
   "abc": 12345,
   "deep": {
    "id": 45
   }
  },
  "type": "removed"
 },
 {
  "key": "group3",
  "value": {
   "deep": {
    "id": {
     "number": 45
    }
   },
   "fee": 100500
  },
  "type": "added"
 }
] 1`] = `
"[
 {
  "type": "nested",
  "key": "common",
  "children": [
   {
    "type": "added",
    "key": "follow",
    "value": false
   },
   {
    "type": "unchanged",
    "key": "setting1",
    "value": "Value 1"
   },
   {
    "type": "removed",
    "key": "setting2",
    "value": 200
   },
   {
    "type": "changed",
    "key": "setting3",
    "value1": true,
    "value2": null
   },
   {
    "type": "added",
    "key": "setting4",
    "value": "blah blah"
   },
   {
    "type": "added",
    "key": "setting5",
    "value": {
     "key5": "value5"
    }
   },
   {
    "type": "nested",
    "key": "setting6",
    "children": [
     {
      "type": "nested",
      "key": "doge",
      "children": [
       {
        "type": "changed",
        "key": "wow",
        "value1": "",
        "value2": "so much"
       }
      ]
     },
     {
      "type": "unchanged",
      "key": "key",
      "value": "value"
     },
     {
      "type": "added",
      "key": "ops",
      "value": "vops"
     }
    ]
   }
  ]
 },
 {
  "type": "nested",
  "key": "group1",
  "children": [
   {
    "type": "changed",
    "key": "baz",
    "value1": "bas",
    "value2": "bars"
   },
   {
    "type": "unchanged",
    "key": "foo",
    "value": "bar"
   },
   {
    "type": "changed",
    "key": "nest",
    "value1": {
     "key": "value"
    },
    "value2": "str"
   }
  ]
 },
 {
  "type": "removed",
  "key": "group2",
  "value": {
   "abc": 12345,
   "deep": {
    "id": 45
   }
  }
 },
 {
  "type": "added",
  "key": "group3",
  "value": {
   "deep": {
    "id": {
     "number": 45
    }
   },
   "fee": 100500
  }
 }
]"
`;
